src/main/java/uk/gov/ch/developer/docs/controller/developer/SignInController.java

```diff
@@ -0,0 +1,231 @@
+import com.nimbusds.jose.JWEObject;
+import com.nimbusds.jose.Payload;
+import com.nimbusds.jose.crypto.DirectEncrypter;
+import net.minidev.json.JSONObject;
+import uk.gov.companieshouse.environment.EnvironmentReader;
+import uk.gov.companieshouse.environment.impl.EnvironmentReaderImpl;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+import uk.gov.companieshouse.session.Session;
+import uk.gov.companieshouse.session.SessionImpl;
+import uk.gov.companieshouse.session.SessionKeys;
+import uk.gov.companieshouse.session.handler.SessionHandler;
+import uk.gov.companieshouse.session.model.SignInInfo;
+import uk.gov.companieshouse.session.model.UserProfile;
+
+@Controller
+@RequestMapping("/signin")
+public class SignInController {
+
+    private static final String OAUTH_COMPANY_SCOPE_PREFIX =
+            "https://api.companieshouse.gov.uk/company/";
+
+    private static final EnvironmentReader reader = new EnvironmentReaderImpl();
+
+    private static final String base64Key = reader.getMandatoryString("OAUTH2_REQUEST_KEY");
+    private static final String authorizationUri = reader.getMandatoryString("OAUTH2_AUTH_URI");
+    private static final String clientId = reader.getMandatoryString("OAUTH2_CLIENT_ID");
+    private static final String redirectUri = reader.getMandatoryString("OAUTH2_REDIRECT_URI");
+    private static final String secret = reader.getMandatoryString("COOKIE_SECRET");
+
+    protected Logger LOGGER = LoggerFactory.getLogger("docs.developer.ch.gov.uk");
+
+    @GetMapping
+    public void getSignIn(ServletRequest request, ServletResponse response) throws IOException {
+
+        Session chSession =
+                (Session) request.getAttribute(SessionHandler.CHS_SESSION_REQUEST_ATT_KEY);
+
+        HttpServletRequest httpServletRequest = (HttpServletRequest) request;
+        HttpServletResponse httpServletResponse = (HttpServletResponse) response;
+
+        // Redirect for user authentication (no scope specified)
+        redirectForAuth(chSession, httpServletRequest, httpServletResponse, null, false);
+
+    }
+
+    /**
+     * Redirects to a URI for the user to authenticate themselves
+     * 
+     * @param session The user's session, retrieved from context
+     * @param request
+     * @param response
+     * @param companyNumber The company number a user is viewing
+     * @param force Boolean determining whether to force re-authentication
+     * @throws IOException
+     */
+    protected void redirectForAuth(Session session, HttpServletRequest request,
+            HttpServletResponse response, String companyNumber, boolean force) throws IOException {
+
+        // Find the original requested url
+        StringBuilder originalRequestUrl = new StringBuilder(request.getRequestURL().toString());
+        String queryString = request.getQueryString();
+        if (queryString != null)
+            originalRequestUrl.append("?").append(queryString);
```

Comment:
Coding standards - always use braces for `if` statements, so that code is consistent and bugs aren't introduced by insertion of additional code. 

Please use the code formatting plugin to enforce CH standards.

Reply:
Zain I know you marked this as resolved but I don't actually see the braces that David wanted.

Reply:
Don't know what happened there, I'll push the change up again.